cogs5e/funcs/lookupFuncs.py

```diff
@@ -25,31 +26,98 @@
 
 class Compendium:
     def __init__(self):
-        self.cfeats = self.load_json('srd-classfeats.json', [])
-        self.classes = self.load_json('srd-classes.json', [])
-        self.conditions = self.load_json('conditions.json', [])
-        self.feats = self.load_json('srd-feats.json', [])
-        self.itemprops = self.load_json('itemprops.json', {})
-        self.monsters = self.load_json('srd-bestiary.json', [])
-        self.names = self.load_json('names.json', [])
-        self.rules = self.load_json('rules.json', [])
-
-        self.spells = [Spell.from_data(r) for r in self.load_json('srd-spells.json', [])]
-        self.backgrounds = [Background.from_data(b) for b in self.load_json('srd-backgrounds.json', [])]
-        self.items = [i for i in self.load_json('srd-items.json', []) if i.get('type') is not '$']
+        self.backgrounds = []
+        self.cfeats = []
+        self.classes = []
+        self.conditions = []
+        self.fancyraces = []
+        self.feats = []
+        self.itemprops = {}
+        self.items = []
+        self.monster_mash = []
+        self.monsters = []
+        self.names = []
+        self.rfeats = []
+        self.rules = []
+        self.spells = []
+        self.srd_backgrounds = []
+        self.srd_items = []
+        self.srd_races = []
+        self.srd_spells = []
+        self.subclasses = []
+
+    async def reload_task(self, mdb=None):
+        wait_for = int(os.getenv('RELOAD_INTERVAL', '300'))  # TODO: decide if 5 minutes is a reasonable default
+        if wait_for > 0:
+            log.info("Reloading data every {} seconds", wait_for)
+            while True:
+                await self.reload(mdb)
+                await asyncio.sleep(wait_for)
+
+    async def reload(self, mdb=None):
```

Comment:
Just declaring the reload function async doesn't actually make it run in an executor by default - to do that, you'd want to make all the blocking calls a sync function (specifically L64-77 and `load_common()`), then call `loop.run_in_executor()`.

Check out utils/functions.py L500 (`generate_token()`) for an example

Reply:
I miss so many Github notifications, oops.

The reload function should only ever be run in an executor, I... think? It's called from reload_task() which is started in bot.py as `bot.loop.create_task(compendium.reload_task(bot.mdb))`, and also called from a bot command. If there's extra work to do then Python async is even weirder than I thought!

Reply:
Oh dear, it is weirder than I thought.